pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -181,26 +205,84 @@ func (ic *ifConfigurator) createContainerLink(endpointName string, result *curre
 	return hnsEP, nil
 }
 
+func (ic *ifConfigurator) getInterfacesConfigByPods(pods sets.String) map[string]*interfacestore.InterfaceConfig {
+	interfaces := make(map[string]*interfacestore.InterfaceConfig)
+	ic.epCache.Range(func(key, value interface{}) bool {
+		ep, _ := value.(*hcsshim.HNSEndpoint)
+		ifConfig := parseOVSPortInterfaceConfigFromHNSEndpoint(ep)
+		if ifConfig == nil {
+			return true
+		}
+		namespacedName := k8s.NamespacedName(ifConfig.PodNamespace, ifConfig.PodName)
+		if pods.Has(namespacedName) {
+			interfaces[namespacedName] = ifConfig
+		}
+		return true
+	})
+	return interfaces
+}
+
+func parseOVSPortInterfaceConfigFromHNSEndpoint(ep *hcsshim.HNSEndpoint) *interfacestore.InterfaceConfig {
+	portData := &ovsconfig.OVSPortData{
+		Name:        ep.Name,
+		ExternalIDs: ep.AdditionalParams,
+	}
+	ifaceConfig := ParseOVSPortInterfaceConfig(portData, nil)
+	if ifaceConfig != nil {
+		var err error
+		ifaceConfig.MAC, err = net.ParseMAC(ep.MacAddress)
+		if err != nil {
+			klog.Errorf("Failed to parse MAC address from OVS external config %s: %v", ep.MacAddress, err)
```

Comment:
you are not parsing the MAC address from the OVS external config here, as it would be `nil`. The MAC address is retrieved from the HNSEndpoint (`ep.MacAddress`).

Reply:
Yes, exactly.

Reply:
OK. I got you. Update `OVS external config` to `HNSEndpoint`.